lsp/nls/src/requests/completion.rs

```diff
@@ -29,18 +29,129 @@ use crate::{
 // We follow the path by traversing a term, type or contract which represents a record
 // and stop when there is nothing else on the path
 
+// Type or contract
+pub enum TorC {
+    T(Types),
+    C(RichTerm),
+}
+
+// #[derive(Eq, PartialEq, Debug, Ord, PartialOrd, Default)]
+struct IdentWithMeta {
+    ident: Ident,
+    info: TorC,
+    item: Option<LinearizationItem<Types>>,
+}
```

Comment:
I think we might also want to print the documentation, the merge priorities, etc. In fact, everything that we print on `hover`, which are mostly what's inside a `Metadata`. Have you considered just storing that here, instead of just `TorC`?

```suggestion
struct IdentWithMeta {
    ident: Ident,
    meta: MetaValue,
    item: Option<LinearizationItem<Types>>,
}
```

Reply:
We already have a `MetaValue` present in `item`; the `info` field is used to get extra info when that is not available. Also, we can just use a `Types` to replace `TorC`, like you suggested.